tests/main/snapd-snap-transition/task.yaml

```diff
@@ -0,0 +1,23 @@
+summary: Ensure the snapd snap transition works
+
+# ubuntu-core-18 already has the snapd snap
+# FIXME: ubuntu-core-16 needs special code for the transition
+systems: [-ubuntu-core-18-*, -ubuntu-core-16-*]
+
+execute: |
+    echo "Ensure no snapd snap is installed"
+    ! snap list snapd
+
```

Comment:
Could we pre-populate the cache with snapd.snap for this to be "fast" and not have the 20 * 6 sleep sequence?

Reply:
Thanks for this suggestion. Let me understand the concern, is it that the test is noisy when it fails? Or the time it takes to download the snapd snap during the test?

Reply:
Tests that "check-sleep-repeat" tend to either over-sleep, over-spam or under-wait. If we populate the cache there will be no uncertain element anymore. I think we could even change it to not wait in this sense at all.

Reply:
I looked into this and I think something like http://paste.ubuntu.com/p/Vsb9hNJZt9/ can do this. In the future this will be easier when "snap download" talks to snapd and caches downloads automatically. I also wonder if we have recent enough hashlib everywhere to support sha3_384. I can do a followup with this.